packages/cli-kit/src/public/node/notifications-system.ts

```diff
@@ -0,0 +1,235 @@
+import {getCurrentCommandId} from './global-context.js'
+import {fileExists, readFile} from './fs.js'
+import {outputDebug} from './output.js'
+import {CLI_KIT_VERSION} from '../common/version.js'
+import {NotificationsKey, cacheRetrieveOrRepopulate, getCache, setCache} from '../../private/node/conf-store.js'
+
+const URL = 'https://raw.githubusercontent.com/Shopify/cli/notifications/notifications.json'
+
+export interface Notifications {
+  notifications: Notification[]
+}
+
+export interface Notification {
+  id: string
+  message: string
+  cta?: {
+    label: string
+    url: string
+  }
+  type: 'info' | 'warning' | 'error'
+  title?: string
+  minVersion?: string
+  maxVersion?: string
+  minDate?: string
+  maxDate?: string
+  commands?: string[]
+  surface?: 'app' | 'theme' | 'hydrogen' | string
+  frequency?: 'always' | 'once' | 'once_a_day' | 'once_a_week'
+}
+
+/**
+ * Shows notifications to the user if they meet the criteria specified in the notifications.json file.
+ *
+ * @param currentSurfaces - The surfaces present in the current project (usually for app extensions).
+ * @returns - A promise that resolves when the notifications have been shown.
+ */
+export async function showNotificationsIfNeeded(currentSurfaces?: string[]): Promise<void> {
+  try {
+    const notifications = await getNotifications()
+    const commandId = getCurrentCommandId()
+    const notificationsToShow = filterNotifications(notifications.notifications, commandId, currentSurfaces)
+    await renderNotifications(notificationsToShow)
+    // eslint-disable-next-line no-catch-all/no-catch-all
+  } catch (error: unknown) {
+    outputDebug('Error retrieving notifications')
+  }
+}
+
+/**
+ * Renders the notifications to the user.
+ *
+ * @param notifications - The notifications to render.
+ */
+async function renderNotifications(notifications: Notification[]) {
+  notifications.forEach((notification) => {
+    const content = {
+      headline: notification.title,
+      body: notification.message,
+      link: notification.cta,
+    }
+    switch (notification.type) {
+      case 'info': {
+        renderInfo(content)
+        break
+      }
+      case 'warning': {
+        renderWarning(content)
+        break
+      }
+      case 'error': {
+        renderError(content)
+      }
+    }
+    setCache(`notification-${notification.id}`, new Date().getTime().toString())
+  })
+}
+
+/**
+ * Get notifications list from cache or fetch it if not present.
+ *
+ * @returns A Notifications object.
+ */
+export async function getNotifications(): Promise<Notifications> {
+  const cacheKey: NotificationsKey = `notifications-${URL}`
+  const rawNotifications = await cacheRetrieveOrRepopulate(cacheKey, fetchNotifications, 24 * 3600 * 1000)
+  return JSON.parse(rawNotifications)
+}
+
+/**
+ * Fetch notifications from GitHub.
+ */
+async function fetchNotifications(): Promise<string> {
+  const response = await fetch(URL)
```

Comment:
There's an analytics helper to separate out network time from time with the command being active. We should wrap this up in that.